File: yeahyak/frontend/src/types/product.ts
export interface Product {
  id: number;
  productName: string;
  productCode: string;
  manufacturer: string;
  type: Type;
  subCategory: SubCategory;
  details?: string;
  unit: string;
  unitPrice: number;
  createdAt: string;
  productImgUrl?: string;
}

export type Type = '전문의약품' | '일반의약품' | '의약외품';

export const TYPES: Type[] = ['전문의약품', '일반의약품', '의약외품'];

export const SUB_CATEGORY_MAP = {
  전문의약품: [
    '항생제',
    '고혈압 치료제',
    '당뇨병 치료제',
    '진통·소염제',
    '정신신경용제',
    '항암제',
    '기타',
  ],
  일반의약품: ['감기약', '소화제', '해열진통제', '지사제', '외용제', '멀미약', '기타'],
  의약외품: ['마스크', '손소독제', '밴드·반창고', '체온계', '구강청결제', '방역용품', '기타'],
} as const;

export type SubCategoryMap = typeof SUB_CATEGORY_MAP;

export type SubCategory = SubCategoryMap[Type][number];

export type SubCategoryWithAll = '전체' | SubCategory;

File: backend/src/main/java/com/yeahyak/backend/entity/Product.java
package com.yeahyak.backend.entity;

import com.yeahyak.backend.entity.enums.MainCategory;
import com.yeahyak.backend.entity.enums.SubCategory;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "Products")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long productId;

    private String productName;
    private String productCode;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private MainCategory mainCategory;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private SubCategory subCategory;

    private String manufacturer;
    @Lob
    private String details;
    private String unit;
    private BigDecimal unitPrice;
    private Boolean isNarcotic;

    @CreationTimestamp
    private LocalDateTime createdAt;

    @Column(length = 1024)
    private String productImgUrl; //약품 이미지 URL
}
